build-system/tasks/check-sourcemaps.js

```diff
@@ -113,9 +113,15 @@ function checkSourcemapMappings(sourcemapJson, map) {
     throw new Error('Could not find mappings array');
   }
 
-  // Zeroth sub-array corresponds to ';' and has no mappings.
-  // See https://www.npmjs.com/package/sourcemap-codec#usage
-  const firstLineMapping = decode(sourcemapJson.mappings)[1][0];
+  // See https://www.npmjs.com/package/sourcemap-codec#usage.
+  const firstLineMapping = decode(sourcemapJson.mappings)[
+    // In closure builds there is a newline immediately after the AMP_CONFIG.
+    // This whole segment has no mapping to the original source.
+    // Therefore we must skip the zeroth sub-array, indicated by a ';'.
+
+    // In esbuild compiled binaries there is no newline after the config, so the first line is fair game.
+    shouldUseClosure() ? 1 : 0
+  ][0];
```

Comment:
Suggestion for part b) of https://github.com/ampproject/amphtml/pull/36105#discussion_r713240645:
```suggestion
  // In closure builds there is a newline immediately after the AMP_CONFIG.
  // This whole segment has no mapping to the original source.
  // Therefore we must skip the zeroth sub-array, indicated by a ';'.
  // In esbuild compiled binaries there is no newline after the config, so the
  // first line is fair game.
  const firstLineIndex = shouldUseClosure() ? 1 : 0;

  // See https://www.npmjs.com/package/sourcemap-codec#usage.
  const firstLineMapping = decode(sourcemapJson.mappings)[firstLineIndex][0];

```

Reply:
Sure! I'll make this change

Reply:
Done!